rohmu/object_storage/s3.py

```diff
@@ -35,6 +34,37 @@
 import time
 
 
+# botocore typing stubs are incomplete. We either have to write all the stubs we need
+# or work around the problem a bit by importing the typing library only for type checking purposes.
+def create_s3_client(
+    *,
+    session: botocore.session.Session,
+    config: botocore.config.Config,
+    aws_access_key_id: Optional[str],
+    aws_secret_access_key: Optional[str],
+    aws_session_token: Optional[str],
+    region_name: str,
+    verify: Optional[bool | str] = None,
+    endpoint_url: Optional[str] = None,
+) -> Any:
```

Comment:
```suggestion
) -> S3Client:
```

and import it somewhere on toplevel if TYPE_CHECKING. With Any here, mypy won't help.

Reply:
ah right... I got what you wanted :rocket: